src/Operation/FlatMap.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Operation;
+
+use Closure;
+use Generator;
+use Iterator;
+
+/**
+ * @template TKey
+ * @template T
+ */
+final class FlatMap extends AbstractOperation
+{
+    /**
+     * @return Closure(callable(T, TKey, Iterator<TKey, T>): T): Closure(Iterator<TKey, T>): Generator<TKey, T>
+     */
+    public function __invoke(): Closure
+    {
+        return
+            /**
+             * @param callable(T, TKey, Iterator<TKey, T>): T $callback
+             */
+            static function (callable $callback): Closure {
+                /** @var Closure(Iterator<TKey, T>): Generator<TKey, T> $flatMap */
+                $flatMap = Pipe::of()(
+                    Map::of()($callback),
+                    Unwrap::of()
```

Comment:
`Unwrap::of()` is basically a shortcut to `Flatten::of()(1)`, maybe we should use it to get an extremelly minor perf improvement? WDYT ?

Reply:
hmm sure I can use that, I didn't think about the performance 😄

Reply:
excellent :+1: